extensions/sql-database-projects/src/controllers/projectController.ts

```diff
@@ -32,11 +45,158 @@ export class ProjectsController {
 
 		// Read datasources.json (if present)
 		const dataSourcesFilePath = path.join(path.dirname(projectFile.fsPath), constants.dataSourcesFileName);
-		newProject.dataSources = await dataSources.load(dataSourcesFilePath);
+
+		try {
+			newProject.dataSources = await dataSources.load(dataSourcesFilePath);
+		}
+		catch (err) {
+			if (err instanceof dataSources.NoDataSourcesFileError) {
+				// TODO: prompt to create new datasources.json; for now, swallow
+			}
+			else {
+				throw err;
+			}
+		}
+
+		this.refreshProjectsTree();
+	}
+
+	public async createNewProject(newProjName: string, newProjUri: vscode.Uri) {
+		const macroDict: Record<string, string> = {
+			'PROJECT_NAME': newProjName,
+			'PROJECT_GUID': '00000000-0000-0000-0000-000000000000'//Guid.create().toString() // TODO: extension building problems when using this library?
+		};
+
+		let newProjFileContents = this.macroExpansion(newSqlProjectTemplate, macroDict);
+
+		let newProjFileName = newProjName;
+
+		if (!newProjFileName.toLowerCase().endsWith(constants.sqlprojExtension)) {
+			newProjFileName += constants.sqlprojExtension;
+		}
+
+		const newProjFilePath = path.join(newProjUri.fsPath, newProjFileName);
```

Comment:
we need some tests on if someone put just the project name vs path (relative or absolute)

Reply:
Put the name or path for what?  I've renamed the parameter to be more clear what its intention is to and disambiguate between the two.  Are you thinking someone might enter the full path of their sqlproj-to-be for name, and still select a folder in the dialog?